packages/events/src/router/event/event.get.test.ts

```diff
@@ -189,3 +196,239 @@ test('Returns event with all actions', async () => {
     )
   ).toHaveLength(2)
 })
+
+describe('Event indexing behavior', () => {
+  let client: ReturnType<typeof createTestClient>
+  let generator: Awaited<ReturnType<typeof setupTestCase>>['generator']
+  let user: Awaited<ReturnType<typeof setupTestCase>>['user']
+
+  beforeEach(async () => {
+    vi.mocked(indexEvent).mockClear()
+    const setup = await setupTestCase()
+    user = setup.user
+    generator = setup.generator
+    client = createTestClient(user)
+  })
+
+  const createEvent = async () => client.event.create(generator.event.create())
+
+  const declareEvent = async (event: CreatedEvent) =>
+    client.event.actions.declare.request(
+      generator.event.actions.declare(event.id)
+    )
+
+  const findCreateAction = (event: CreatedEvent) =>
+    event.actions.find((action) => action.type === ActionType.CREATE) ??
+    ({} as { createdBy: string })
+
+  const assignEvent = async (event: CreatedEvent, assignedTo: string) =>
+    client.event.actions.assignment.assign({
+      ...generator.event.actions.assign(event.id, { assignedTo }),
+      transactionId: getUUID()
+    })
+
+  const validateEvent = async (event: CreatedEvent) =>
+    client.event.actions.validate.request(
+      generator.event.actions.validate(event.id)
+    )
+
+  const registerEvent = async (event: CreatedEvent) =>
+    client.event.actions.register.request(
+      generator.event.actions.register(event.id)
+    )
+
+  const archiveEvent = async (event: CreatedEvent) =>
+    client.event.actions.archive.request(
+      generator.event.actions.archive(event.id)
+    )
+
+  const rejectEvent = async (event: CreatedEvent) =>
+    client.event.actions.reject.request(
+      generator.event.actions.reject(event.id)
+    )
+
+  const printCertificate = async (event: CreatedEvent) =>
+    client.event.actions.printCertificate.request(
+      generator.event.actions.printCertificate(event.id)
+    )
+
+  const correctionRequest = async (event: CreatedEvent) =>
+    client.event.actions.correction.request.request(
+      generator.event.actions.correction.request(event.id)
+    )
+
+  const correctionApprove = async (event: CreatedEvent, actionId: string) =>
+    client.event.actions.correction.approve.request(
+      generator.event.actions.correction.approve(event.id, actionId)
+    )
+
+  const correctionReject = async (event: CreatedEvent, actionId: string) =>
+    client.event.actions.correction.reject.request(
+      generator.event.actions.correction.reject(event.id, actionId, {
+        reason: { message: 'No legal proof' }
+      })
+    )
```

Comment:
Could we have used the [createEvent](https://github.com/opencrvs/opencrvs-core/blob/develop/packages/events/src/tests/utils.ts#L296) function?